fix(core): re-read the fence before the queue-empty assert in Device::maintain (#9958)
gfx-rs/wgpu: CHANGELOG.md

- Zero-initialize padding (if any) at the end of a buffer allocation. This was application-visible in rare cases on Vulkan when a shader read beyond the valid range of a vertex buffer. By @andyleiserson in [#9791](https://github.com/gfx-rs/wgpu/pull/9791).
- Fix required immediate slots calculation and remove `naga::valid::FunctionInfo::immediate_slots_used`. By @beicause in [#9725](https://github.com/gfx-rs/wgpu/pull/9725).

> - Fix a panic on the queue-empty assert in `Device::maintain` when more than one thread polls the same device. A timed-out `Wait` poll that races a concurrent poll on another thread now returns `WaitSucceeded` or `Timeout` instead of `QueueEmpty`; `QueueEmpty` is only returned when the poll itself proved it, as its documentation already promised. Polling again reports the empty queue. By @AdrianEddy in [#9958](https://github.com/gfx-rs/wgpu/pull/9958).

Reviewer: ```suggestion
- Fix a spurious assertion failure in `Device::maintain` when multiple threads race polling the same device. By @AdrianEddy in [#9958](https://github.com/gfx-rs/wgpu/pull/9958).
```
